File: frontend/src/services/api/collections.ts
// Optimization V2 (M3.9)：Collection Definition API 模块。
// 新增 API 全部走独立模块，避免继续膨胀 services/api.ts（计划书第 21 节）。
import { http } from '@/services/api'

export interface CollectionDefinition {
  id: string
  case_id: string
  version: number
  status: 'draft' | 'active' | 'superseded'
  goal: string
  platforms: string[]
  platform_queries: Record<string, string[]>
  exclusions: string[]
  filters: Record<string, unknown>
  generated_by_run_id: string | null
  created_at: string
  updated_at: string
}

export interface CollectionDefinitionPayload {
  goal: string
  platforms: string[]
  platform_queries?: Record<string, string[]>
  exclusions?: string[]
  filters?: Record<string, unknown>
}

export const collectionApi = {
  async list(caseId: string): Promise<CollectionDefinition[]> {
    const { data } = await http.get<CollectionDefinition[]>(
      `/cases/${caseId}/collection-definitions`,
    )
    return data
  },
  async getActive(caseId: string): Promise<CollectionDefinition | null> {
    try {
      const { data } = await http.get<CollectionDefinition>(
        `/cases/${caseId}/collection-definitions/active`,
      )
      return data
    } catch (error) {
      const status = (error as { response?: { status?: number } }).response?.status
      if (status === 404) return null
      throw error
    }
  },
  async create(
    caseId: string,
    payload: CollectionDefinitionPayload,
  ): Promise<CollectionDefinition> {
    const { data } = await http.post<CollectionDefinition>(
      `/cases/${caseId}/collection-definitions`,
      payload,
    )
    return data
  },
  async generate(
    caseId: string,
    goal?: string,
  ): Promise<CollectionDefinition> {
    const { data } = await http.post<CollectionDefinition>(
      `/cases/${caseId}/collection-definitions:generate`,
      { goal: goal ?? null },
    )
    return data
  },
  async revise(
    caseId: string,
    definitionId: string,
    payload: Partial<CollectionDefinitionPayload>,
  ): Promise<CollectionDefinition> {
    const { data } = await http.post<CollectionDefinition>(
      `/cases/${caseId}/collection-definitions/${definitionId}:revise`,
      payload,
    )
    return data
  },
  async activate(caseId: string, definitionId: string): Promise<CollectionDefinition> {
    const { data } = await http.post<CollectionDefinition>(
      `/cases/${caseId}/collection-definitions/${definitionId}:activate`,
    )
    return data
  },
}

File: backend/app/api/routes/collections.py
"""M3: Collection Definition routes（采集定义版本化 API）。"""

from __future__ import annotations

from fastapi import APIRouter, Depends, status

from app.api.dependencies import get_container
from app.bootstrap import ApplicationContainer
from app.core.errors import ApplicationError
from app.schemas.collections import (
    CollectionDefinitionResponse,
    CreateCollectionDefinitionRequest,
    GenerateCollectionDefinitionRequest,
    ReviseCollectionDefinitionRequest,
)

router = APIRouter()


@router.get("/{case_id}/collection-definitions", response_model=list[CollectionDefinitionResponse])
async def list_collection_definitions(
    case_id: str,
    container: ApplicationContainer = Depends(get_container),
) -> list[CollectionDefinitionResponse]:
    records = await container.collection_service.list_for_case(case_id)
    return [CollectionDefinitionResponse.from_record(record) for record in records]


@router.get(
    "/{case_id}/collection-definitions/active",
    response_model=CollectionDefinitionResponse,
)
async def get_active_collection_definition(
    case_id: str,
    container: ApplicationContainer = Depends(get_container),
) -> CollectionDefinitionResponse:
    record = await container.collection_service.get_active(case_id)
    if record is None:
        raise ApplicationError(
            "no active collection definition", code="collection_not_found"
        )
    return CollectionDefinitionResponse.from_record(record)


@router.post(
    "/{case_id}/collection-definitions",
    response_model=CollectionDefinitionResponse,
    status_code=status.HTTP_201_CREATED,
)
async def create_collection_definition(
    case_id: str,
    request: CreateCollectionDefinitionRequest,
    container: ApplicationContainer = Depends(get_container),
) -> CollectionDefinitionResponse:
    record = await container.collection_service.create_manual(
        case_id,
        goal=request.goal,
        platforms=request.platforms,
        platform_queries=request.platform_queries,
        exclusions=request.exclusions,
        filters=request.filters,
    )
    return CollectionDefinitionResponse.from_record(record)


@router.post(
    "/{case_id}/collection-definitions:generate",
    response_model=CollectionDefinitionResponse,
    status_code=status.HTTP_201_CREATED,
)
async def generate_collection_definition(
    case_id: str,
    request: GenerateCollectionDefinitionRequest,
    container: ApplicationContainer = Depends(get_container),
) -> CollectionDefinitionResponse:
    record = await container.collection_service.generate(
        case_id, goal=request.goal
    )
    return CollectionDefinitionResponse.from_record(record)


@router.post(
    "/{case_id}/collection-definitions/{definition_id}:revise",
    response_model=CollectionDefinitionResponse,
    status_code=status.HTTP_201_CREATED,
)
async def revise_collection_definition(
    case_id: str,
    definition_id: str,
    request: ReviseCollectionDefinitionRequest,
    container: ApplicationContainer = Depends(get_container),
) -> CollectionDefinitionResponse:
    record = await container.collection_service.revise(
        case_id,
        definition_id,
        goal=request.goal,
        platforms=request.platforms,
        platform_queries=request.platform_queries,
        exclusions=request.exclusions,
        filters=request.filters,
    )
    return CollectionDefinitionResponse.from_record(record)


@router.post(
    "/{case_id}/collection-definitions/{definition_id}:activate",
    response_model=CollectionDefinitionResponse,
)
async def activate_collection_definition(
    case_id: str,
    definition_id: str,
    container: ApplicationContainer = Depends(get_container),
) -> CollectionDefinitionResponse:
    record = await container.collection_service.activate(case_id, definition_id)
    return CollectionDefinitionResponse.from_record(record)
